app/controllers/administrateurs/types_de_champ_controller.rb

```diff
@@ -61,6 +61,24 @@ def move
       draft.move_type_de_champ(params[:stable_id], params[:position].to_i)
     end
 
+    def move_and_morph
+      source_type_de_champ = draft.find_and_ensure_exclusive_use(params[:stable_id])
+      target_type_de_champ = draft.find_and_ensure_exclusive_use(params[:target_stable_id])
+      @coordinate = @source_coordinate = draft.coordinate_for(source_type_de_champ)
```

Comment:
tu sembles ne pas avoir besoin de `source_coordinate`

```suggestion
      @coordinate = draft.coordinate_for(source_type_de_champ)
```